File: subarray-division.py
def birthday(s, d, m):
    count = 0
    n = len(s)
    for i in range(n-m+1): 
        total = 0
        for j in range(i,i+m):
            total += s[j]
        if total == d:
            count += 1
    return count

File: subarray-division.java
class Result {
    public static int birthday(List<Integer> s, int d, int m) {
        int count = 0;
        int n = s.size();
        for(int i = 0;i<=n-m;i++){
            int sum = 0;
            for(int j = i;j<i+m;j++){
                sum += s.get(j);
            }
            if(sum == d){
                count++;
            }
        }
        return count;
    }
}